confer.go

```diff
@@ -178,6 +184,19 @@ func (manager *Config) Get(key string) interface{} {
 	return v
 }
 
+// NewConfigSubset returns a new config based on
+func (manager *Config) NewConfigSubset(prefix string) *Config {
+	configSubset := &Config{}
+	*configSubset = *manager
```

Comment:
Interesting. You're choosing to copy the data here rather than maintaining a reference. Would it surprise you to find that your subset was unchanged after modifying a value in the parent config?